tests/test_tablemanager.py

```diff
@@ -0,0 +1,177 @@
+  'voxel_resolution_z': 40.0,
+  'live_compatible': False,
+  'voxel_resolution_y': 4.0},
+ 'single_neurons': {'notice_text': None,
+  'datastack_name': 'minnie65_phase3_v1',
+  'id': 5,
+  'voxel_resolution_x': 4.0,
+  'description': 'This is a list of all the single neuron objects in the dataset, the id column is the nucleus_id, the root_id is its segment_id (which can change over versions), and then position is the position of the centroid of its nucleus. This is based on the soma_counts view, filtering for cell_type="neuron" and n_neuron=1. ',
+  'voxel_resolution_z': 40.0,
+  'live_compatible': False,
+  'voxel_resolution_y': 4.0}}
+
+views_schema = {'synapses_pni_2_in_out_degree': {'root_id': {'type': 'integer'},
+  'n_input': {'type': 'integer'},
+  'avg_input_size': {'type': 'float'},
+  'n_output': {'type': 'integer'},
+  'avg_output_size': {'type': 'float'}},
+ 'connections_with_nuclei': {'pre_pt_root_id': {'type': 'integer'},
+  'post_pt_root_id': {'type': 'integer'},
+  'n_syn': {'type': 'integer'},
+  'sum_size': {'type': 'float'},
+  'pre_nuc_id': {'type': 'integer'},
+  'post_nuc_id': {'type': 'integer'}},
+ 'nucleus_detection_lookup_v1': {'id': {'type': 'integer'},
+  'volume': {'type': 'float'},
+  'pt_position': {'type': 'SpatialPoint'},
+  'pt_root_id': {'type': 'integer'},
+  'orig_root_id': {'type': 'integer'},
+  'pt_supervoxel_id': {'type': 'integer'},
+  'pt_position_lookup': {'type': 'SpatialPoint'}},
+ 'soma_counts': {'id': {'type': 'integer'},
+  'pt_position': {'type': 'SpatialPoint'},
+  'pt_root_id': {'type': 'integer'},
+  'pt_supervoxel_id': {'type': 'integer'},
+  'cell_type': {'type': 'string'},
+  'neuron_count': {'type': 'integer'},
+  'total_count': {'type': 'integer'}},
+ 'single_neurons': {'id': {'type': 'integer'},
+  'pt_root_id': {'type': 'integer'},
+  'pt_position': {'type': 'SpatialPoint'},
+  'pt_supervoxel_id': {'type': 'integer'}}}
+
+def test_table_manager():
+    tables = TableManager(None, metadata=table_meta, schema=schema_meta)
+    qry = tables.allen_column_mtypes_v2(pt_root_id=[123, 456], target_id=271700)
+    params = qry.filter_kwargs_live
+    assert len(tables)==2
+    assert 'allen_column_mtypes_v2' in params.get('filter_equal_dict')
+    assert 'nucleus_detection_v0' in params.get('filter_in_dict')
+    assert 'allen_column_mtypes_v2' == qry.joins_kwargs.get('joins')[0][0]
+
+def test_view_manager():
+    views = ViewManager(None, view_metadata=views_list, view_schema=views_schema)
```

Comment:
this should be done with responses so it tests that portion of the code

Reply:
```        with ThreadPoolExecutor(max_workers=4) as executor:
            metadata.append(
                executor.submit(
                    self.get_tables_metadata,
                )
            )
            metadata.append(
                executor.submit(
                    self.fc.schema.schema_definition_all
                )
            )
            metadata.append(
                executor.submit(
                    self.get_views
                )
            )
            metadata.append(
                executor.submit(
                    self.get_view_schemas
                )
            )
        if self.fc is not None:
            tables = TableManager(self.fc, metadata[0].result(), metadata[1].result())
        else:
            tables = None
        self.tables = tables
```

This is how the code is supplied to table manager now. I could do this with responses, but won't I just be plugging the answer directly in anyway?

Reply:
Fair, changed now.